src/ngraph/pass/constant_folding.cpp

```diff
@@ -234,6 +234,12 @@ void pass::ConstantFolding::construct_constant_reshape()
                          fold_constant_reshape<double>(constant_match, reshape_match, func));
             return true;
         }
+        else if (type == element::bf16)
+        {
+            replace_node(m.get_match_root(),
+                         fold_constant_reshape<int16_t>(constant_match, reshape_match, func));
```

Comment:
I don't think it makes any difference but it would certainly read better if you used ngraph::bfloat16 here